core/Notification/Manager.php

```diff
@@ -112,6 +113,17 @@ private static function addNotification($id, Notification $notification)
         $session->notifications[$id] = $notification;
     }
 
+    private static function removeOldestNotificationsIfThereAreTooMany()
+    {
+        $maxNotificationsInSession = 30;
+
+        $session = static::getSession();
+
+        while (count($session->notifications) >= $maxNotificationsInSession) {
+            array_shift($session->notifications);
+        }
```

Comment:
Of course it's possible but bit more complex since we want to remove the first once. I had a `array_slice` first but prefer it this way